.github/workflows/release.yml

```diff
@@ -0,0 +1,122 @@
+name: Release
+
+on:
+  workflow_dispatch:            # allow manually triggering this workflow
+  push:
+    tags:
+    - "v*"
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  RUST_BACKTRACE: 1
+  RUSTFLAGS: -Dwarnings
+  RUSTDOCFLAGS: -Dwarnings
+  MSRV: "1.72"
+  SCCACHE_CACHE_SIZE: "50G"
+
+jobs:
+    build_release:
+        timeout-minutes: 60
+        name: Build release binaries
+        runs-on: ${{ matrix.runner }}
+        continue-on-error: false
+        strategy:
+          fail-fast: false
+          matrix:
+            name: [ubuntu-latest, ubuntu-arm-latest, macOS-latest, macOS-arm-latest]
+            rust: [stable]
+            experimental: [false]
+            include:
+              - name: ubuntu-arm-latest
+                os: ubuntu-latest
+                release-os: linux
+                release-arch: aarch64
+                runner: [self-hosted, linux, ARM64]
+              - name: ubuntu-latest
+                os: ubuntu-latest
+                release-os: linux
+                release-arch: amd64
+                runner: [self-hosted, linux, X64]
+              - name: macOS-latest
+                os: macOS-latest
+                release-os: darwin
+                release-arch: x86_64
+                runner: [self-hosted, macOS, X64]
+              - name: macOS-arm-latest
+                os: macOS-latest
+                release-os: darwin
+                release-arch: aarch64
+                runner: [self-hosted, macOS, ARM64]
+              # - os: windows-latest
+              #   release-os: windows
+              #   release-arch: amd64
+              #   runner: [windows-latest]
+        env:
+          # Using self-hosted runners so use local cache for sccache and
+          # not SCCACHE_GHA_ENABLED.
+          RUSTC_WRAPPER: "sccache"
+          RUST_BACKTRACE: full
+          RUSTV: ${{ matrix.rust }}
+        steps:
+        - name: Checkout
+          uses: actions/checkout@master
+          with:
+            submodules: recursive
+    
+        - name: Set build arch
+          run: |
+            echo "RELEASE_ARCH=${{ matrix.release-arch }}" >> $GITHUB_ENV
+            echo "RELEASE_OS=${{ matrix.release-os }}" >> $GITHUB_ENV
+    
+        - name: Install ${{ matrix.rust }}
+          run: |
+            rustup toolchain install ${{ matrix.rust }}
+    
+        - name: Install sccache
+          uses: mozilla-actions/sccache-action@v0.0.3
```

Comment:
Yeah, this is mostly a hoping to get some deps filled since the release builds run so long. But generally I don't have high hopes for cache doing anything here. Let's see how next release rolls and we can drop it to at least save the disk space :D